### DISCUSSION TITLE
React and Feliz support for extended enterprise usage

### DISCUSSION THREAD

**Discussion by jkone27** [Lanayx/Oxpecker#28] [General]
idea: suggest if wanting to make FE side code appetible, to target also React/Next.js as those are kind of relevant frameworks today, 

this is why Fable.React and Feliz exist and i think providing a common template/integration with Feliz/React would be very welcome useful? 

i understand this new CE/DSL is smarter/better than list feliz DSL, but then i think having an adapter making the 2 compatibles would be great for extending existing Feliz or Feliz.ViewEngine apps, and especially for apps using Fable.React `ReactElement` as output?

https://hygraph.com/blog/javascript-frameworks

https://www.statista.com/statistics/1124699/worldwide-developer-survey-most-used-frameworks-web/

* https://fable.io/blog/2022/2022-10-12-react-jsx.html
* https://zaid-ajaj.github.io/Feliz/
* https://github.com/fable-compiler/Feliz.JSX/tree/master?tab=readme-ov-file

**Comment by Lanayx** [comment]
> idea: suggest if wanting to make FE side code appetible, to target also React/Next.js as those are kind of relevant frameworks today,

If somebody wants to work on Oxpecker.React and it's Fable plugin, I'll approve that, not going to work on it myself. My initial idea was to create a new framework (similar to Sutil), but it so happened that Solid already is a framework that fulfills my needs, so I just created bindings not to reinvent the wheel

> this is why Fable.React and Feliz exist and i think providing a common template/integration with Feliz/React would be very welcome useful?

I don't know, I currently see them as a competitive choices to make.

> i understand this new CE/DSL is smarter/better than list feliz DSL, but then i think having an adapter making the 2 compatibles would be great for extending existing Feliz or Feliz.ViewEngine apps, and especially for apps using Fable.React ReactElement as output?

It's possible to extend Oxpecker.Solid to accept JSX.Element (used in Feliz.JSX.Solid) as a child or parameter, however I don't think there are many projects using Feliz.JSX.Solid, so I don't see benefits of doing that (not very trivial) work

**Comment by shayanhabibi** [comment]
The way I see it, we don't really have much specific in the plugins that wouldn't produce correct React JSX right?

Mappings for loops inside children is probably the only thing.

If there is still interest, I could look at this in a month or two. If possible, I could probably make it so there is just a react bindings library which uses the Oxpecker.Solid.FablePlugin. That library would just need to extend the HtmlContainer computation with the For method.

However, I don't know about Lanayx, but I certainly wouldn't be interested either in *maintaining* the bindings. If this is something you are interested in (@jkone27 ) then I can help you author the bindings/plugin.

My only interest is spreading Lanayxs DSL 💇
